keps/993-two-phase-admission/README.md

```diff
@@ -253,6 +254,27 @@ The controller implementing a particular check should:
 * After approving the workload, keep an eye on the check if it starts failing,
   fail the check and cause the workload to move back to the suspended state.
 
+### Preemption Admission Check Controller
+
+Because now, the time when the eviction of the preemption candidate should vary based on the preemptor state
+the scheduler will not issue the eviction during it's process instead it will set a `Pending` admission check
+that is manged by a new built-in admission check controller.
+
+The **Admission Check Controller** will:
+
+- Watch for a change in state of the workloads pending preemption.
+- Watch for workloads that are finishing execution and therefore releasing quota.
+- Watch for AdmissionCheck changes, since the preemption policy can change.
+
+Since evaluating the current preemption state and the needs of a workload requires checking the sate of all resource pool the events are rate limited the kueue's cache is used.
+
+At every run the controller will get the list of workloads pending preemption.
+Since for some of these workloads is not necessary to issue eviction at that given point (eg. Having a pending check that uses AfterCheckPassedOrOnDemand policy) their quota reservation will be ignored.
+Fore every other preemption pending workloads, it will check if it can fit without evicting other workloads, case in which the preemption admission check condition will be set to `Ready`.
```

Comment:
The quota reserved by a workload is considered  in use by the scheduler, if another workload gets is reservation after the fist one gets to "preemption ready" , that second workload will need to trigger it's own set of evictions and so on.

Reply:
(fore -> for done)

Reply:
> The quota reserved by a workload is considered in use by the scheduler, if another workload gets is reservation after the fist one gets to "preemption ready" , that second workload will need to trigger it's own set of evictions and so on.

Yes, that I understand.

Let me state my thinking again. There are scenarios where preemption needs to run before other admission checks have completed, because other AdmissionChecks can request preemptions.
In that case, we will issue preemptions, but even when completed, we might still need preemptions later on, when all the AdmissionChecks actually complete.

That's why I say that there might not be a concept of "preemptions are ready".

Reply:
The preemption is finished when the workload can fit without the need of any additional evictions. Pass this point its  quota is locked and can start execution, or wait more.  The quote remains locked until this workload finishes or gets evicted.

Reply:
That makes me think that the existing `QuotaReserved` condition will have different semantics. And the idea of `QuotaReserved` could actually be used to signal that preemption is "ready".

But maybe that's a bit of over-engineering.

Reply:
Would QuotaReserved be equivalent to "preemption ready" then?